app/models/message.rb

```diff
@@ -16,6 +16,21 @@ def mark_as_deleted(user)
     self.recipient_status = 'd' if recipient == user
   end
 
+  def text # rubocop:disable Metrics/AbcSize
+    case param_type
+      when 'group'
+        I18n.t('groups.send_access_request_message.message', user: sender.name, group: Group.find(param_id).name)
+      when 'group_accepted'
+        I18n.t('groups.send_grant_access_messages.message', user: sender.name, group: Group.find(param_id).name)
+      when 'group_declined'
+        I18n.t('groups.send_deny_access_message.message', user: sender.name, group: Group.find(param_id).name)
+      when 'collection'
+        I18n.t('collections.share_message.text', user: sender.name, collection: Collection.find(param_id).title)
+      else
+        super
+    end
+  end
+
```

Comment:
Maybe you have another idea here? When we store the preferred locale for a user in the database we could also use this to translate the messages before they get written to the database and then remove this part here.

Reply:
Indeed, we could store the translated message already. However, I would argue that this approach has some drawbacks, since the message is then "fixed". For example, the message won't be updated to another language when the user switches the locale. Further, we cannot update the message either, e.g., when rephrasing the message text. Hence, I would probably keep your current solution, since it is more flexible.